src/backend-api/Controllers/AddController.cs

```diff
@@ -514,6 +509,229 @@ public IActionResult PostProgram([FromBody] PostProgramInputModel input)
+		                "EndOfLife" : string? (formatted yyyy-mm-dd),
+		                "EmployeeId" : int?,
+		                "TextField" : string?,
+		                "ScreenSize" : float?,
+		                "CostPerYear" : decimal?,
+		                "Resolution" : decimal?,
+		                "Mfg" : string?,
+		                "Make" : string?,
+		                "Model" : string?,
+		                "Fqdn" : string?,
+		                "Location" : "xx"? (either GR or AA),
+		                "SerialNumber" : string?,
+		                "MonthsPerRenewal" : int?
+	                }
+                }
+         * Return: 201 created if successful, and 400 bad request if not.
+         */
+        [HttpPost]
+        [Route("Laptop")]
+        [Route("Computer")]
+        public IActionResult PostComputer([FromBody] EntityInput<Computer> input)
+        {
+            return PostHardware(input.Entity, _context.Computer);
+        }
+
+        /* POST: api/add/peripheral
+         * Method will add a peripheral row to the peripheral table with the specified
+         *   attribute values if they are valid and will add the appropriate hardware history.
+         * Input format:
+                {
+	                "Entity" : {
+		                "PeripheralName" : string?,
+		                "PeripheralType" : string?,
+		                "TextField" : string?,
+		                "EmployeeId" : int?,
+		                "FlatCost" : decimal?,
+		                "PurchaseDate" : string? (formatted yyyy-mm-dd),,
+		                "CostPerYear" : decimal?,
+		                "Mfg" : string?,
+		                "Location" : "xx"? (either GR or AA),
+		                "RenewalDate" : string? (formatted yyyy-mm-dd),
+		                "SerialNumber" : string?,
+		                "MonthsPerRenewal" : int?,
+	                }
+                }
+         * Return: 201 created if successful, and 400 bad request if not.
+         */
+        [HttpPost]
+        [Route("Peripheral")]
+        public IActionResult PostPeripheral([FromBody] EntityInput<Peripheral> input)
+        {
+            return PostHardware(input.Entity, _context.Peripheral);
+        }
+
+        /* PostHardware<T>(hardware, table) is a method to post any hardware type 
+         *   to it's corresponding table and add hardware history.
+         * Return: 201 created if successful, and 400 bad request if not.
+         */
+        private IActionResult PostHardware<T>(T hardware, DbSet<T> table)
+            where T : class, IHardwareBase
+        {
+            int? EmployeeId = hardware.EmployeeId;
+
+            // Update values we don't want touched by the endpoint call.
+            hardware.IsAssigned = EmployeeId != null ? true : false;
+            hardware.IsDeleted = false;
+
+            // Get class name at runtime
+            string type = GetClassName(hardware);
+
+            try
+            {
+                table.Add(hardware);
```

Comment:
How are we managing monthsPerRenewal controlling the isCostPerYear

Reply:
Delegated to front-end